src/query/src/dist_plan/commutativity.rs

```diff
@@ -68,8 +69,8 @@ impl Categorizer {
                 }
 
                 // sort plan needs to consider column priority
-                // We can implement a merge-sort on partial ordered data
-                Commutativity::PartialCommutative
+                // Change Sort to MergeSort which assume the input streams is already sorted hence can be more efficient
+                Commutativity::ConditionalCommutative(Some(Arc::new(merge_sort_transformer)))
```

Comment:
We should ensure the number of partition is not smaller than the number of region at present. Otherwise this would result in incorrect output.